rust/connlib/clients/android/src/lib.rs

```diff
@@ -194,6 +194,16 @@ impl Callbacks for CallbackHandler {
                     name: "dns_addresses",
                     source,
                 })?;
+            let search_domain = search_domain
+                .map(|domain| {
+                    env.new_string(domain.to_string())
+                        .map_err(|source| CallbackError::NewStringFailed {
+                            name: "search_domain",
+                            source,
+                        })
+                })
+                .transpose()?
+                .unwrap_or_default();
```

Comment:
Does this return a nullable pointer if `None`? Otherwise I think this might send an empty string by default which won't be what we want on the other side of the FFI right?

Reply:
Yes the default for `JString` is null.

Reply:
I can add a quick unit test for that.

Reply:
Nevermind, scratch that - just tested and I think the `Option` makes it over the FFI. I wouldn't have thought so, quite surprising.

Reply:
> Nevermind, scratch that - just tested and I think the `Option` makes it over the FFI. I wouldn't have thought so, quite surprising.

The `Option` doesn't, the function-signature is still `Ljava/lang/String;`, so we are passing a `String`. But everything in Java can be nullable so we are effectively passing null. The `Default` impl for `JString` creates a `JObject::null`, which is why this works.